official/vision/beta/dataloaders/classification_input.py

```diff
@@ -141,24 +156,26 @@ def _parse_eval_data(self, decoded_tensors):
 
   def _parse_train_image(self, decoded_tensors):
     """Parses image data for training."""
+    # TODO: add option to crop images
     image_bytes = decoded_tensors[self._image_field_key]
-    image_shape = tf.image.extract_jpeg_shape(image_bytes)
+    if 'image/height' in decoded_tensors and 'image/width' in decoded_tensors:
+      image_shape = (decoded_tensors['image/height'], decoded_tensors['image/width'], 3)
+    else:
+      image_shape = tf.image.extract_jpeg_shape(image_bytes)
 
-    # Crops image.
-    # TODO(pengchong): support image format other than JPEG.
-    cropped_image = preprocess_ops.random_crop_image_v2(
-        image_bytes, image_shape)
-    image = tf.cond(
-        tf.reduce_all(tf.equal(tf.shape(cropped_image), image_shape)),
-        lambda: preprocess_ops.center_crop_image_v2(image_bytes, image_shape),
-        lambda: cropped_image)
+    image = tf.io.decode_image(image_bytes, channels=3)
+    image = tf.reshape(image, image_shape)
```

Comment:
do you have to do this since the scale is 1 and preserve aspect is true?

Reply:
The reshaping is required to give the decoded tensor the correct shape before manipulating.